backend/locator/api.py

```diff
@@ -1,12 +1,109 @@
 import datetime
 import flask
+import sqlalchemy as sa
 
-from   . import model
+from   .model import Event
+
+# FIXME: We keep deleted events (deleted=true) but don't keep revisions of
+# patched events.  To do this, we'd have to manage event IDs ourselves rather
+# than relying on an autoincrement key.
 
 #-------------------------------------------------------------------------------
 
+class APIError(Exception):
+
+    def __init__(self, message, status_code=400):
+        self.message = message
+        self.status_code = status_code
+
+
+
+def jsonify(jso, status_code=200):
```

Comment:
this is fine, but Flask provides a Response class & object that allow you to do just this (Response(resp, status_code)) in place of jsonify